sqlstats: remove p99, p90, p50 latencies from sql stats

dhartunian opened the pull request:
## sqlstats: remove p99, p90, p50 latencies from sql stats

Previously, we would report latencies for sql statements as part of
our stats collection. The implementation has proved to be confusing
to understand and provides little utility to customers in its current
form. This is due to the fact that the aggregation intervals for
the latencies don't match the aggregation timestamps for the other
sql stats, and our aggregation logic cannot properly combine latency
observations together because the full histogram is not stored.

For now, we are disabling the collection and reporting of
these percentiles to avoid confusion on the user side. We will
likely re-introduce these stats in the future with a different
implementation.

Resolves: #126912

Release note (ui change): Percentile latencies are no longer
available for SQL Activity. The implementation of these percentiles
was error-prone and difficult to understand since it was computed
differently from the other SQL statistics we collect. Customers
interested in viewing percentile latencies per statement fingerprint
are encouraged to use the experimental per-fingerprint histograms that
can be enabled with the `sql.stats.detailed_latency_metrics.enabled`
cluster setting. This will enable externalized histogram metrics via
the Prometheus scrape endpoint.

----
## sql: make detailed latency metric setting public

Previously, this setting was not documented. Given that we're
disabling the usage of percentile metrics for SQL Activity, we want to
point customers to alternatives more easily.

The `sql.stats.detailed_latency_metrics.enabled` cluster setting
is now publicly documented and marked as experimental and affecting
performance

Release note: None

dhartunian committed:
sqlstats: remove p99, p90, p50 latencies from sql stats

Previously, we would report latencies for sql statements as part of
our stats collection. The implementation has proved to be confusing
to understand and provides little utility to customers in its current
form. This is due to the fact that the aggregation intervals for
the latencies don't match the aggregation timestamps for the other
sql stats, and our aggregation logic cannot properly combine latency
observations together because the full histogram is not stored.

For now, we are disabling the collection and reporting of
these percentiles to avoid confusion on the user side. We will
likely re-introduce these stats in the future with a different
implementation.

Resolves: #126912

Release note (ui change): Percentile latencies are no longer
available for SQL Activity. The implementation of these percentiles
was error-prone and difficult to understand since it was computed
differently from the other SQL statistics we collect. Customers
interested in viewing percentile latencies per statement fingerprint
are encouraged to use the experimental per-fingerprint histograms that
can be enabled with the `sql.stats.detailed_latency_metrics.enabled`
cluster setting. This will enable externalized histogram metrics via
the Prometheus scrape endpoint.

dhartunian committed:
sql: make detailed latency metric setting public

Previously, this setting was not documented. Given that we're
disabling the usage of percentile metrics for SQL Activity, we want to
point customers to alternatives more easily.

The `sql.stats.detailed_latency_metrics.enabled` cluster setting
is now publicly documented and marked as experimental and affecting
performance

Release note: None

xinhaoz reviewed (approved):
LGTM after CI passes - I think some tests are failing because you might need to update tests checking the deleted `crdb_internal.x` columns.

dhartunian committed:
sqlstatsccl: simplify config in TestSQLStatsRegions

Hypothesis: all the cluster setting customizations in this test
are causing flakes and lockups but making subsystems behave in
non-standard ways.

I ran the test a few times without any settings changes and it seems
to run in about the same time or faster than before. It appears that
their intention was to make the test "faster" but that doesn't seem to
be the case, especially since we don't care to run it under duress.

Resolves: #139600
Release note: None

dhartunian commented:
TFTR!

bors r=xinhaoz
